### DISCUSSION TITLE
Migrating from Sqflite

### DISCUSSION THREAD

**Discussion by aleripe** [powersync-ja/sqlite_async.dart#17] [Q&A] [answered]
Hi, it's me again! After converting all my data layer I was trying to test the migration on an existing database.
I'm currently using _sqflite_ and my database is at version 24 (I don't really know where this value is stored as I can't find it in the database).
The problem is when I run my updated app with _sqlite_async_ (_sqlite3_) it tries to execute all the migrations as it thinks database is at version 0 but it fails finding the tables already exist. I see this library creates a migration table where it stores all the values.

Has someone already tried to migrate from _sqflite_ to _sqlite_async_ (_sqlite3_)? Thanks!

EDIT: _sqflite_ stores database version in _PRAGMA user_version_

**Comment by aleripe** [comment]
I don't know if this method could be modified to check _PRAGMA user_version_ if _$migrationTable_ has not been found...

```
Future<int> getCurrentVersion(SqliteWriteContext db) async {
    try {
      final currentVersionRow = await db.getOptional(
          'SELECT ifnull(max(id), 0) as version FROM $migrationTable');
      int currentVersion =
          currentVersionRow == null ? 0 : currentVersionRow['version'];
      return currentVersion;
    } on SqliteException catch (e) {
      if (e.message.contains('no such table')) {
        return 0;
      }
      rethrow;
    }
```
  }

**Reply by rkistner** [reply, marked as answer]
The migrations in sqlite_async stores more than just the version number, and `PRAGMA user_version` could have been used for different purposes, so we can't do that check in the core library.

However, you can add a check directly to the app's migrations, such as:

```dart
final migrations = SqliteMigrations()
  ..add(SqliteMigration(1, (tx) async {
    final userVersion = (await db.get('PRAMGA user_version'))['user_version'];
    if (userVersion < 1) {
      await tx.execute(
          'CREATE TABLE test_data(id INTEGER PRIMARY KEY AUTOINCREMENT, data TEXT)');
    } else {
       // Already migrated in sqflite, ignore
    }
  }))
```

**Reply by aleripe** [reply]
Thanks a lot Ralf, this is a good suggestion. I'll try to see if this suits my needs, otherwise I'll have to fork your library. Thanks again! Cheers
